src/engine.ts

```diff
@@ -288,13 +292,14 @@ export class Engine<A extends string = never, T extends Task<A> = Task<A>> {
    * @param task The current executing task.
    */
   do(task: T): void {
+    const macroString = Macro.load().toString();
     if (typeof task.do === "function") {
       task.do();
```

Comment:
Maybe we should pass the combat macro string along here as well, if the task.do wants to handle some of the combat?